### DISCUSSION TITLE
Adding a pdbConverter

### DISCUSSION THREAD

**Discussion by dessygil** [polaris-hub/polaris#104] [Ideas]
Hi all,

I'm currently trying to add the dataset suggested in a previous [discussion post](https://github.com/polaris-hub/polaris/discussions/87) called [PDE10A](https://pubmed.ncbi.nlm.nih.gov/36153472/). In this dataset, there are various files representing the ligand, pocket and protein. The ligand is within a .sdf file, whereas the pocket and protein are represented in a .pdb or .mol2 file. Polaris already has the function to convert a .sdf file to be used but I would like to add a .pdb converter to add the functionality to Polaris and to use for the specific dataset I'm trying to add.

In short, I'm writing this post to discuss the best strategy for adding the pdb converter.

**Comment by zhu0619** [comment]
Hi @dessygil, 
Thank you for your participation and idea sparking.

One idea we have but haven't explored much is to convert PDB to NumPy matrices and store them in .zarr format, similar to the SDF converter. Came across this [library](https://github.com/samuelmurail/pdb_numpy), which could be a good starting point.

CC @cwognum

**Reply by cwognum** [reply]
That looks like a nice library! I like that it only depends on `numpy` and `cython`.

But maybe [`Bio.PDB`](https://biopython.org/docs/1.75/api/Bio.PDB.html) is the more mature package here? BioPython also has just a single dependency I believe (i.e., `numpy`). If the worry is that with all of BioPython we're pulling in a lot of complexity and code that we won't need, I believe we could also install just the `Bio.PDB` subpackage, but I would have to double check that!

**Reply by dessygil** [reply]
Hey @zhu0619 @cwognum, 

of course! Bio.PDB is the one I've been looking to use already but let me know if there is any other library you would want instead

**Comment by cwognum** [comment]
Besides parsing the PDB file, one question is how to save the data contained in a PDB file to an array. This would ideally be lossless if possible, but remember that Polaris is ultimately meant for ML modeling. Instead of lossless, it would thus also suffice to just save the information needed for modeling. Maybe as a first step, we could simply store the 3D coordinates and amino acid sequence? Maybe worth going through a couple of ML papers that actually cite [PDE10A](https://pubmed.ncbi.nlm.nih.gov/36153472/) and see which information they use?

**Reply by dessygil** [reply]
I'll check some papers tomorrow after work and list the papers checked and the data used in this thread

**Reply by dessygil** [reply]
Some papers that discuss how the PDE10A dataset was used.

[Exploring protein-ligand binding affinity prediction with electron density-based geometric deep learning](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10835705/#cit62)
- Nothing explicit about which files were used from the PDE10A dataset. 
- Bond-critical points and their quantum mechanical properties require coordinates, elements, and connectivity. 

[Expanding Training Data for Structure-Based Receptor–Ligand Binding Affinity Regression through Imputation of Missing Labels](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10634251/)
- Used the PDE10A dataset to perform docking experiments using Gnina
- For docking, it says online data from the PDB file that's required is atom records (coordinates, elements, connectivity), chain information, and ligand information

[EQUIVARIANT SCALAR FIELDS FOR MOLECULAR DOCKING WITH FAST FOURIER TRANSFORMS](https://openreview.net/pdf?id=BIveOmD1Nh)
- They are using the PDE10A dataset for docking,
- The same information would be required as the article above

Let me know your thoughts. I can only access so many papers since I can only access what's free but it seems like the we should follow the information that the docking experiments need as everything else would then use a subset of that information if anything.

**Comment by zhu0619** [comment]
Hi, @dessygil 
Just wanted to check in and see how things are going with pdb converter. :)

**Comment by cwognum** [comment]
This suggestion has been implemented in https://github.com/polaris-hub/polaris/pull/171 ! 🎉
